pypsa/linopf.py

```diff
@@ -1038,5 +1040,12 @@ def save_optimal_capacities(n, iteration, status):
     n.links[['p_nom', 'p_nom_extendable']] = n.links['p_nom_opt'], False
     kwargs['warmstart'] = False
     network_lopf(n, snapshots, **kwargs)
+    if track_iterations:
+        n.global_constraints = n.global_constraints.rename(columns={'mu':f'mu_{iteration}'})
+    if 'lv_limit' in n.global_constraints.index:
+        logger.info(f'Resulting dual value for line volume limit (lv_limit) set to the one of iteration {iteration-1} '
+                    'due to its optimisation process.') # otherwise, final dual for lv_limit is NaN.
+        if track_iterations: n.global_constraints.at['lv_limit', f'mu_{iteration}'] = final_duals['lv_limit']
+        else: n.global_constraints.at['lv_limit', f'mu'] = final_duals['lv_limit']
```

Comment:
Wouldn't you want to do this in any case (when iterating) not just if you track iterations?

Reply:
Not sure I get your question. The name of the column in n.global_constraints is different, that's why I added the if/else, and the logger.info is displayed in any case. But if we track iterations and want to remain with NaN for the last iteration, this part will be different.